pkg/apis/k3k.io/v1alpha1/types.go

```diff
@@ -40,6 +41,13 @@ type ClusterList struct {
 	Items []Cluster `json:"items"`
 }
 
+type PersistenceConfig struct {
+	// Type can be ephermal, static, dynamic
+	Type               string `json:"type"`
```

Comment:
It might be nice to attach a `.validate() error` method to this type to check if the given value in the `Type` field is a valid one and to short circuit if it's not.